packages/core/src/shared/telemetry/spans.ts

```diff
@@ -364,6 +375,34 @@ export class TelemetryTracer extends TelemetryBase {
         }
     }
 
+    /**
+     * Trace represents a higher level event that you want to link with any other spans. It generates
+     * a traceId which is added to all subsequent calls, allowing you to query for all related events.
+     *
+     * Functionally, it is similiar to run. The only difference is the traceId gets added to subsequent telemetry calls.
```

Comment:
Why not just replace `run` completely with `trace` (or add the login to run directly). We often won't know to make a `run` a `trace` until its too late, and it would be easy for folks to use forget or not know to use `trace`.

Reply:
I think they recommend against in opentelemetry but I think it might make sense here since we have so many starting/ending flows vs traditional software

Reply:
how do you decided end-to-end of traceId in user journey flow? can you give an example

Reply:
A traceId is just a uuid. If its the most top level telemetry event in a flow it creates the traceId. Otherwise it adds the traceId from the parent metric.

For example, a vscode_executeCommand can contain contain code that does amazonq_enterFocusChat, amazonq_openChat.

When vscode_executeCommand runs it generates trace id (which is a random uuid e.g.): 'aaaaa-aaaaa-aaaaa-aaaaa-aaaaa' this trace id gets attached to amazonq_enterFocusChat, amazonq_openChat so we get the final 3 metrics:

vscode_executeCommand:
traceId: 'aaaaa-aaaaa-aaaaa-aaaaa-aaaaa'

amazonq_enterFocusChat
traceId: 'aaaaa-aaaaa-aaaaa-aaaaa-aaaaa'

amazonq_openChat
traceId: 'aaaaa-aaaaa-aaaaa-aaaaa-aaaaa'

then in our telemetry instance you can look up traceId: 'aaaaa-aaaaa-aaaaa-aaaaa-aaaaa' and see all the events associated with that particular run. The next time vscode_executeCommand gets ran a new traceId is generated and then attached to the subsequent metrics. It's just following the ideas from https://opentelemetry.io/docs/concepts/signals/traces/